cmd/egress/cmd.go

```diff
@@ -6,29 +6,17 @@ import (
 	"os"
 	"time"
 
-	"github.com/aws/aws-sdk-go-v2/credentials"
 	ocmlog "github.com/openshift-online/ocm-sdk-go/logging"
 	"github.com/openshift/osd-network-verifier/pkg/cloudclient"
 	"github.com/spf13/cobra"
 )
 
 var (
-	defaultTags            = map[string]string{"osd-network-verifier": "owned", "red-hat-managed": "true", "Name": "osd-network-verifier"}
-	regionEnvVarStr string = "AWS_REGION"
-	regionDefault   string = "us-east-2"
+	defaultTags     = map[string]string{"osd-network-verifier": "owned", "red-hat-managed": "true", "Name": "osd-network-verifier"}
+	regionEnvVarStr = "AWS_REGION"
+	regionDefault   = "us-east-2"
 )
 
-type egressConfig struct {
```

Comment:
This struct is defined in cloudclient and fields required by AWS are added to AWS client to remove need for cyclic imports

Reply:
This struct should stay local to the egress command and only be used here. The cloudclient should hvae its own struct with only options that apply to creating the client.

Reply:
It's not possible to implement it that way without cycling imports. The imports get deeper and import cycles develop. It's counter intuitive but this is what came out of a few days of trying multiple ways. I'm open to refactor later on.

Reply:
What cyclic imports were happening?

Reply:
I'm -1 to refactoring later. This is a major change to how this api works and will be difficult to unwind.

Reply:
It's not possible. We are aiming to use struct instead of vars to pass to the client, so the struct must be defined in the client. If it is defined in cmd, 2 major things are wrong with that. 
  1. Downstream code has to define the struct, so we are not providing them a streucture for input. 
  2. cloudclient must import egress command and go does not like the cyclic import. We shouldn't be importing downstream code.

Reply:
I'm with you on -1 to that refactor as it won't work.

Reply:
The struct for the AWS client should be dfined in the AWS code. The struct to hold options for the specific test should be help in the test definition. There should be two different structs. The code in egress.go should build the struct needed for AWS Client (or GCP client) And pass it to the NewClient function. There should be no cyclic imports there

Reply:
Tried that, still had issues..

Reply:
What issues specifically?

Reply:
I wrote an explanation https://github.com/openshift/osd-network-verifier/pull/107#issuecomment-1171778922